flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarOperatorsTest.scala

```diff
@@ -74,6 +74,238 @@ class ScalarOperatorsTest extends ScalarOperatorsTestBase {
       "true")
   }
 
+  @Test
+  def testShiftLeft(): Unit = {
+    testAllApis(
+      3.shiftLeft(3),
+      "3.shiftLeft(3)",
+      "SHIFTLEFT(3,3)",
+      "24"
+    )
+
+    testAllApis(
+      2147483647.shiftLeft(-2147483648),
+      "2147483647.shiftLeft(-2147483648)",
+      "SHIFTLEFT(2147483647,-2147483648)",
+      "2147483647"
+    )
+
+    testAllApis(
+      -2147483648.shiftLeft(2147483647),
+      "-2147483648.shiftLeft(2147483647)",
+      "SHIFTLEFT(-2147483648,2147483647)",
+      "0"
+    )
+
+    testAllApis(
+      9223372036854775807L.shiftLeft(-2147483648),
+      "9223372036854775807L.shiftLeft(-2147483648)",
+      "SHIFTLEFT(9223372036854775807,-2147483648)",
+      "9223372036854775807"
+    )
+
+    testAllApis(
+      'f3.shiftLeft(5),
+      "f3.shiftLeft(5)",
+      "SHIFTLEFT(f3,5)",
+      "32"
+    )
+
+    testAllApis(
+      1.shiftLeft(Null(Types.INT)),
+      "1.shiftLeft(Null(INT))",
+      "SHIFTLEFT(1, CAST(NULL AS INT))",
+      "null"
+    )
+
+    testAllApis(       // test tinyint
```

Comment:
regarding the tinyint, smallint, int and bigint tests. I think we need the following tests that show the quirky nature of java bit shifts:
```
select cast(1 as tinyint) << 9, cast(1 as tinyint) << 17;
select cast(1 as smallint) << 17, cast(1 as smallint) << 33;
select 1 << 17, 1 << 33,
select cast(1 as bigint) << 33, cast(1 as bigint) << 65;
```

expected results:
```
0, 2
0, 2
131072, 2
8589934592, 2
```
Bonus points for anyone that understands those results 😜 

Also we need the same for both versions of right shifts, but instead of right shifting `1`, right shift min values (`Byte.MIN_VALUE`, `Short.MIN_VALUE`, ...) (and results should also be different)

Reply:
Thanks for review and suggestions, @pnowojski . I updated the codes according to review.